pkg/sql/catalog/descs/collection.go

```diff
@@ -81,10 +81,23 @@ type Collection struct {
 
 	uncommittedZoneConfigs uncommittedZoneConfigs
 
-	// A collection of descriptors which mirrors the descriptors committed to
-	// storage. This acts as a cache by accumulating every descriptor ever read
-	// from KV in the transaction affiliated with this Collection.
-	stored catkv.StoredCatalog
+	// A cached implementation of catkv.CatalogReader used for accessing stored
+	// descriptors, namespace entries, comments and zone configs. The cache
+	// accumulates every descriptor and other catalog data ever read from KV
+	// in the transaction affiliated with this Collection.
+	cr catkv.CatalogReader
+
+	// shadowedNames is the set of name keys which should not be looked up in
+	// storage. Maintaining this set is necessary to properly handle the renaming
+	// of descriptors, especially when attempting to reuse an old name: without
+	// this set the check that verifies that the target name is not already in
+	// use would systematically fail.
+	shadowedNames map[descpb.NameInfo]bool
```

Comment:
nit: I understand that we never set the value to `false` for any key, so we don't need to worry about whether a key does not exist or the value is `false`. But I think the convention is use `struct{}` as the value type.

Reply:
Yes, this rings a bell. I'll adjust.

Reply:
a friendly reminder that this is not fixed. But I don't hold a strong opinion blocking this. Feel free to ignore.

Reply:
Sorry! Thanks will do.

Reply:
Done.